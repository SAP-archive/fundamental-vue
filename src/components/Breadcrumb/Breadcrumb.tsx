import { DefaultSlot, Component, Base } from '@/core';

@Component('Breadcrumb')
@DefaultSlot('Breadcrumb Items')
export class Breadcrumb extends Base {
  public render() {
    return <ul class='fd-breadcrumb'>{this.$slots.default}</ul>;
  }
}
