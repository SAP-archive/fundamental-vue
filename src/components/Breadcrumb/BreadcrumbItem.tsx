import { warn, Component, Event, DefaultSlot, Prop, Base } from '@/core';

interface Props {
  to?: object | null;
}

@Component('BreadcrumbItem')
@Event('click', 'Sent when item was clicked', ['item', 'BreadcrumbItem'])
@DefaultSlot('Breadcrumb Item Title')
export class BreadcrumbItem extends Base<Props> {
  @Prop('target route (passed to $router.to(â€¦))', {
    type: Object,
    default: null,
  })
  public to!: object | null;

  private onClick(event: MouseEvent) {
    event.preventDefault();
    const { to, $router } = this;
    if (to != null) {
      if($router != null) {
        $router.push(to);
      } else {
        warn(`Tried to navigate to ${to} but $router not found.`);
      }
    }
    this.$emit('click', this);
  }

  public render() {
    const title = this.$slots.default;
    return (
      <li class='fd-breadcrumb__item'>
        <a class='fd-breadcrumb__link' href='#' on-click={this.onClick}>{title}</a>
      </li>
    );
  }
}
