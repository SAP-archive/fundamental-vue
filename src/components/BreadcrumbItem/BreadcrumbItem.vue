<template>
  <li class="fd-breadcrumb__item">
    <a class="fd-breadcrumb__link" @click="pushLocationIfPossible" v-bind="attrs" v-on="$listeners">
      <!-- Anchor tag content -->
      <slot />
    </a>
  </li>
</template>

<script>
import { withTargetLocation } from "./../../mixins";
// A breadcrumb item renders a single items and is used in combination with `fd-breadcrumb`. All attributes will be passed down to the anchor-tag that is rendered on your behalf.
export default {
  name: "FdBreadcrumbItem",
  inheritAttrs: false,
  mixins: [withTargetLocation()],
  computed: {
    attrs() {
      return {
        href: "#",
        ...this.$attrs
      };
    }
  }
};
</script>
