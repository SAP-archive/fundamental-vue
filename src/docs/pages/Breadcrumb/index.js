export default {
  relatedComponents: ["fd-breadcrumb", "fd-breadcrumb-item"]
};
