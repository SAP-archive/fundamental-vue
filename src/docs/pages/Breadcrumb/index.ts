import { ExampleCollectionFunction } from "../types";

export const plugin: ExampleCollectionFunction = () => {
  return {
    status: "inprogress",
    icon: "navigation-right-arrow",
    related: ["FdBreadcrumb", "FdBreadcrumbItem"]
  };
};
