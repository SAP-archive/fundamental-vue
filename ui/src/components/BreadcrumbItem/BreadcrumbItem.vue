<template>
  <li class="fd-breadcrumb__item">
    <a
      class="fd-breadcrumb__link"
      @click="pushLocationIfPossible"
      v-bind="attrs"
      v-on="$listeners"
    >
      <slot />
    </a>
  </li>
</template>

<script lang="ts">
import { withTargetLocation, mixins } from "@/mixins";

export default mixins(withTargetLocation()).extend({
  name: "FdBreadcrumbItem",
  inheritAttrs: false,
  computed: {
    attrs(): object {
      return {
        href: "#",
        ...this.$attrs
      };
    }
  }
});
</script>
