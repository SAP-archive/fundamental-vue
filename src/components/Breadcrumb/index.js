import Breadcrumb from "./Breadcrumb.vue";
import { pluginify } from "./../../util";
export default pluginify(Breadcrumb);
export { Breadcrumb };
