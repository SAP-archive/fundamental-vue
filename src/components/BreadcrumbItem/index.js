import BreadcrumbItem from "./BreadcrumbItem.vue";
import { pluginify } from "./../../util";
export default pluginify(BreadcrumbItem);
export { BreadcrumbItem };
