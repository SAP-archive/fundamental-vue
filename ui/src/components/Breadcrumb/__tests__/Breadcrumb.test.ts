import { mount } from "@vue/test-utils";
import Breadcrumb from "../Breadcrumb.vue";
import BreadcrumbItem from "../../BreadcrumbItem/BreadcrumbItem.vue";
import { VNode, CreateElement } from "vue";

describe("Breadcrumb", () => {
  it("renders correctly", () => {
    const wrapper = mount(Breadcrumb, {
      slots: {
        default: {
          render(h: CreateElement): VNode {
            return h(
              BreadcrumbItem,
              {
                attrs: {
                  href: "#item1"
                }
              },
              "Item 1"
            );
          }
        }
      }
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
