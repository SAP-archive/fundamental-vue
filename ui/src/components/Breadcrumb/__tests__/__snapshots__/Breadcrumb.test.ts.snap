// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb renders correctly 1`] = `
<ul
  class="fd-breadcrumb"
>
  <div>
    <li
      class="fd-breadcrumb__item"
    >
      <a
        class="fd-breadcrumb__link"
        href="#item1"
      >
        Item 1
      </a>
    </li>
    <li
      class="fd-breadcrumb__item"
    >
      <a
        class="fd-breadcrumb__link"
        href="#item2"
      >
        Item 2
      </a>
    </li>
  </div>
</ul>
`;
